app/models/auction.rb

```diff
@@ -49,6 +49,13 @@ class Auction < ActiveRecord::Base
     format: { with: Regexp.new("#{C2_REGEX}[0-9]") },
     allow_blank: true
   )
+  validate :publishing_auction, on: :update, if: :published_changed?
+
+  def publishing_auction
+    if published_was == 'unpublished' && purchase_card == 'default' && c2_status != 'approved'
```

Comment:
Since we are doing this validation when `:published_changed?`, this is an extra step whether we should validate or not based on the value of `published` field.

For e.g. if you want to unpublish **published** auction, then this validation is just extra work to be done and it will be skipped since `published_was` was "published".